telebot/api.nim

```diff
@@ -594,29 +594,36 @@ proc getUpdates*(b: TeleBot, offset, limit, timeout = 0, allowedUpdates: seq[str
   if allowedUpdates.len > 0:
     data["allowed_updates"] = $allowedUpdates
 
-  let res = await makeRequest(endpoint % b.token, data)
-  for item in res.items:
-    var update = unmarshal(item, Update)
-    if update.updateId > b.lastUpdateId:
-      b.lastUpdateId = update.updateId
+  result = await makeRequest(endpoint % b.token, data)
+
+  if result.len > 0:
+    b.lastUpdateId = result[result.len - 1]{"update_id"}.to(int)
 
-    if update.inlineQuery.isSome:
-      for cb in b.inlineQueryCallbacks:
-        await cb(b, update.inlineQuery.get)
-    elif update.hasCommand():
-      var userCommands = update.getCommands()
-      for command in userCommands.keys():
-        if b.commandCallbacks.hasKey(command):
-          var cmd: Command
-          cmd.message = update.message.get()
-          cmd.params = userCommands[command]
+proc handleUpdate*(b: TeleBot, update: Update) {.async.} =
+  if update.updateId > b.lastUpdateId:
+    b.lastUpdateId = update.updateId
 
-          for cb in b.commandCallbacks[command]:
-            await cb(b, cmd)
-    else:
-      for cb in b.updateCallbacks:
-        await cb(b, update)
+  if update.inlineQuery.isSome:
+    for cb in b.inlineQueryCallbacks:
+      await cb(b, update.inlineQuery.get)
+  elif update.hasCommand():
+    var userCommands = update.getCommands()
+    for command in userCommands.keys():
+      if b.commandCallbacks.hasKey(command):
+        var cmd: Command
+        cmd.message = update.message.get()
+        cmd.params = userCommands[command]
 
+        for cb in b.commandCallbacks[command]:
+          await cb(b, cmd)
+  else:
+    for cb in b.updateCallbacks:
+      await cb(b, update)
+
+proc cleanUpdates*(b: TeleBot) {.async.} =
+  var updates = await b.getUpdates()
+  while updates.len >= 100:
```

Comment:
why 100?

Reply:
from getUpdates api doc "limit: Limits the number of updates to be retrieved. Values between 1—100 are accepted. Defaults to 100."
it could be a variable but then it should be passed in both getUpdates calls which made the code less clean, I can add it if you'd prefer that

Reply:
the idea is to retrieve the all updates till it falls below the max limit, which indicates there are no more updates to fetch

Reply:
i got it, thank you!